Fixes

Grazfather opened the pull request:
More miscellaneous stuff:

* Looked for docstrings that didn't end with a period.
* Looked for us using byte strings when we really don't need to be (e.g. when writing python scripts to disk)
* Looked for us mutating class instances when we could use immutable objects.
* Replaced `pprint` with `psprint`: The only difference is that it returns the string to be printed, instead of printing it outright. Just wanted to keep the responsibility of printing outside of any non-command class ideally.
* Replaced `'` with `"` where it made sense.

hugsy commented:
I'll review this PR this weekend since it's quite easy (low chances to break anything) and you mention that you might push some more stuff.

Grazfather commented on gef.py:
```diff
@@ -4542,7 +4530,7 @@ def create_or_edit_structure(self, mod_name, struct_name):
         if not self.is_valid_struct(mod_name):
             info("Creating '{:s}' from template".format(fullname))
-            with open(fullname, "wb") as f:
+            with open(fullname, "b") as f:
```
Typo. Good eye. Was supposed to be `"w"`, I deleted the wrong char.

Grazfather commented:
>L2640 "Sync" must be capitalized for the RPC to execute correctly

Ok

>L4533 open() without 'w' will trigger an Exception on f.write() (L4534). Why did you remove the 'w'?

Typo, good catch.

> L4545 When I implemented this on Py3, I had to use bytes. Did you actually test this on Py2 and Py3 ?

Yes, you only needed `wb` because you put `b"` before all the strings in `get_template`, which I fixed.

> L4872 Prefer """ to avoid "

Ok

Grazfather commented:
> L2640 "Sync" must be capitalized for the RPC to execute correctly

Actually, I don't understand why this is the case.

hugsy commented:
> > L2640 "Sync" must be capitalized for the RPC to execute correctly
> 
> Actually, I don't understand why this is the case.

Nvm, that used to be how I did it originally (`ida-interact` + `RPC_Method`) which is `Sync` not `sync` (https://github.com/hugsy/gef/blob/master/scripts/ida_gef.py#L200) But it's not the case anymore, so forget what I said...
